### PR TITLE
Update VCS status properly for directories with symlink parents

### PR DESCRIPTION
**Pull request by acontreras89**
Fix https://github.com/atom/tree-view/issues/484

### CODE REVIEW

File: lib/directory.coffee [hunk 1]
```diff
@@ -64,7 +64,7 @@ class Directory
     return unless repo?
 
     @subscriptions.add repo.onDidChangeStatus (event) =>
-      @updateStatus(repo) if event.path.indexOf("#{@path}#{path.sep}") is 0
+      @updateStatus(repo) if event.path.indexOf("#{@realPath}#{path.sep}") is 0
```
Comment (acontreras89) [inline lib/directory.coffee]: Should we keep the old condition as well? For files (which are working fine) both `@path` and `@realPath` are checked ([here](https://github.com/atom/tree-view/blob/master/lib/file.coffee#L53) and [here](https://github.com/atom/tree-view/blob/master/lib/file.coffee#L76).)
I haven't seen `repo`'s `onDidChangeStatus` event containing a path with a symlink directory, but I guess that would be a little safer to do.

**Comment by mnquintana** [comment]
This looks good, but would you mind adding a spec to help prevent regressions in the future? Thanks!

**Comment by acontreras89** [comment]
:+1: There you go @mnquintana

**Comment by acontreras89** [comment]
Hey @mnquintana , could you take a look at this again? :smile:

**Comment by acontreras89** [comment]
It'd be nice if someone could review this :)

**Comment by mnquintana** [comment]
Sorry about that! Been super busy - I'll take a look tonight.

**Comment by mnquintana** [comment]
Awesome, thanks so much for fixing this! :rocket:

**Comment by acontreras89** [comment]
My pleasure!

**Pull request merged**